src/test/java/io/strimzi/kafka/bridge/http/ConsumerTest.java

```diff
@@ -0,0 +1,1116 @@
+            () -> produce.complete(true), () -> new ProducerRecord<>(topic, 0, null, sentBody.getBytes()));
+        produce.get(TEST_TIMEOUT, TimeUnit.SECONDS);
+
+        String baseUri = Urls.consumerInstance(groupId, name);
+
+        JsonObject json = new JsonObject();
+        json.put("name", name);
+        json.put("format", "binary");
+
+        // create consumer
+        // subscribe to a topic
+        consumerService()
+            .createConsumer(context, groupId, json)
+            .subscribeConsumer(context, groupId, name, topic);
+
+        CompletableFuture<Boolean> consume = new CompletableFuture<>();
+        // consume records
+        consumerService()
+            .getRequest(Urls.consumerInstanceRecords(groupId, name, 1000, null))
+                .putHeader("Accept", BridgeContentType.KAFKA_JSON_BINARY)
+                .as(BodyCodec.jsonArray())
+                .send(ar -> {
+                    context.verify(() -> {
+                        assertTrue(ar.succeeded());
+                        HttpResponse<JsonArray> response = ar.result();
+                        assertEquals(HttpResponseStatus.OK.code(), response.statusCode());
+                        JsonObject jsonResponse = response.body().getJsonObject(0);
+
+                        String kafkaTopic = jsonResponse.getString("topic");
+                        int kafkaPartition = jsonResponse.getInteger("partition");
+                        String key = jsonResponse.getString("key");
+                        String value = new String(DatatypeConverter.parseBase64Binary(jsonResponse.getString("value")));
+                        long offset = jsonResponse.getLong("offset");
+
+                        assertEquals(topic, kafkaTopic);
+                        assertEquals(sentBody, value);
+                        assertEquals(0L, offset);
+                        assertNotNull(kafkaPartition);
+                        assertNull(key);
+                    });
+                    consume.complete(true);
+                });
+
+        consume.get(TEST_TIMEOUT, TimeUnit.SECONDS);
+
+        // consumer deletion
+        consumerService()
+            .deleteConsumer(context, groupId, name);
+        context.completeNow();
+        assertTrue(context.awaitCompletion(TEST_TIMEOUT, TimeUnit.SECONDS));
+    }
+
+    @Test
+    void receiveFromMultipleTopics(VertxTestContext context) throws InterruptedException, ExecutionException, TimeoutException {
+        String topic1 = "receiveSimpleMessage-1";
+        String topic2 = "receiveSimpleMessage-2";
+        kafkaCluster.createTopic(topic1, 1, 1);
+        kafkaCluster.createTopic(topic2, 1, 1);
+
+        String sentBody1 = "Simple message-1";
+        String sentBody2 = "Simple message-2";
+
+        CompletableFuture<Boolean> produce = new CompletableFuture<>();
+        kafkaCluster.useTo().produce("", 1, new KafkaJsonSerializer(), new KafkaJsonSerializer(),
```

Comment:
In general, there should be `CompletableFuture` for each producer, this should be fixed. I suggest to factor out native kafka producer/consumer piece of code to methods